### PR TITLE
Consolidate option parsing classes

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

**Comment by milleruntime** [comment]
What are these changes doing?

**Comment by mikewalch** [comment]
> What are these changes doing?

We have a lot of option parsing classes that subclass each other.  It's confusing.  This change reduces the number of these classes and removes code.

**Comment by milleruntime** [comment]
OK I see... you are just removing a bunch of similar CLI classes and replacing them with ClientOpts.

File: server/tracer/src/main/java/org/apache/accumulo/tracer/TraceTableStats.java [hunk 1]
```diff
@@ -33,14 +33,15 @@
...
  * Reads the trace table and prints out some stats about the spans found.
  */
 public class TraceTableStats {
-  static class Opts extends ClientOnDefaultTable {
-    public Opts() {
-      super("trace");
-    }
+  static class Opts extends ClientOpts {
+    @Parameter(names = "--table", description = "table to use")
+    String tableName = "trace";
```
Comment (milleruntime) [inline server/tracer/src/main/java/org/apache/accumulo/tracer/TraceTableStats.java]: Why not just add tableName to ClientOpts?

Comment (mikewalch) [reply server/tracer/src/main/java/org/apache/accumulo/tracer/TraceTableStats.java]: Options don't always require/use a table.

Comment (milleruntime) [reply server/tracer/src/main/java/org/apache/accumulo/tracer/TraceTableStats.java]: Right but can't you set it to null like some of the options that aren't always used?

Comment (mikewalch) [reply server/tracer/src/main/java/org/apache/accumulo/tracer/TraceTableStats.java]: I know you can set a parameter to default to null but that doesn't remove it as option. It'll still print out on the command line if user prints the help.

Comment (milleruntime) [reply server/tracer/src/main/java/org/apache/accumulo/tracer/TraceTableStats.java]: Oh right, I forgot about the Help.

**Pull request merged**